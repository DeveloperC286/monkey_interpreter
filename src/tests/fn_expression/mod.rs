use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case("fn(x) {\n\tx + 3\n}\n", "fn_expression_case_1"),
    case("fn(x,y) { x + y; };", "fn_expression_case_2"),
    case("FN() {\n3\n}\n", "fn_expression_case_3"),
    case("fn(x,y) {\n\treturn x + y;\n}", "fn_expression_case_4"),
    case("fn(){}", "fn_expression_case_5")
)]
fn fn_expression_cases(code: &str, snapshot_name: &str) {}

#[apply(fn_expression_cases)]
fn test_fn_expression_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(fn_expression_cases)]
fn test_fn_expression_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(fn_expression_cases)]
fn test_fn_expression_evaluation(code: &str, snapshot_name: &str) {
    assert_evaluation!(code, snapshot_name);
}

#[apply(fn_expression_cases)]
fn test_fn_expression_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
