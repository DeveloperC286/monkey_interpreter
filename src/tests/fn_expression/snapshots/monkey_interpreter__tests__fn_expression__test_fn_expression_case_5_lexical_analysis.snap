---
source: src/tests/fn_expression/mod.rs
expression: tokens
---
[
    Function,
    OpeningRoundBracket,
    ClosingRoundBracket,
    OpeningCurlyBracket,
    ClosingCurlyBracket,
    EndOfFile,
]
